src/test/java/nextstep/subway/line/application/LineServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package nextstep.subway.line.application;
+
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.domain.LineRepository;
+import nextstep.subway.line.dto.LineRequest;
+import nextstep.subway.line.dto.LineResponse;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.dto.StationResponse;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static nextstep.subway.line.domain.LineFixture.신이호선;
+import static nextstep.subway.line.domain.LineFixture.이호선;
+import static nextstep.subway.station.StationFixture.당산역;
+import static nextstep.subway.station.StationFixture.합정역;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@Service
+public class LineServiceTest {
+
+    LineRepository lineRepository = mock(LineRepository.class);
+    StationService stationService = mock(StationService.class);
+    LineService lineService = new LineService(lineRepository, stationService);
+
+    @DisplayName("노선을 저장한다.")
+    @Test
+    void 노선_저장_테스트() {
+        // given
+        LineRequest lineRequest = new LineRequest("2호선", "Green", 당산역.getId(), 합정역.getId(), 20);
+
+        when(stationService.findById(당산역.getId())).thenReturn(당산역);
+        when(stationService.findById(합정역.getId())).thenReturn(합정역);
+        when(lineRepository.save(lineRequest.toLine(당산역, 합정역))).thenReturn(이호선());
```

Comment:
```suggestion
        given(stationService.findById(당산역.getId())).willReturn(당산역);
        given(stationService.findById(합정역.getId())).willReturn(합정역);
        given(lineRepository.save(lineRequest.toLine(당산역, 합정역))).willReturn(이호선());
```

위에서 드린 링크에도 있는 내용이지만 일반적으로 테스트를 작성할 때 BDD 스타일을 적용합니다.
이해하기 쉬운 시나리오 기반이라서 쓰기 편하거든요.
따라서 when 메소드 대신 BDD 스타일을 지원하는 given 메소드로 교체하시는걸 권장드립니다.
given 단계에서 when 메소드로 given 역할을 수행하는게 조금 이상하잖아요

https://www.baeldung.com/bdd-mockito

Reply:
save를 여기서 mocking 하고 싶다는 것 같은데 맞으실까요?

```java
Line saved = new Line("저장된 노선", "파란색");
ReflectionTestUtils.setField(saved, "id", 1L);
given(lineRepository.save(any())).willReturn(saved);
```

보통 JPA를 사용할 때 ID는 생성자에 넣지 않습니다.
ReflectionTestUtils 을 사용하면 리플렉션으로 ID를 주입시켜줄 수 있어요.
이렇게 `Line` 을 생성하는걸 `LineFixture` 에 넣어두고 사용하시면 편리합니다.

Reply:
아 기존에는 `id` 값을 자동으로 생성했다면 mocking 에서는 해당 값이 없기 때문에 계속 에러가 나던 거였네요!
알려주신 대로 `id값을` 지정하고 `given-willReturn`  을 활용하니 정상적으로 처리되어서 테스트 코드 수정 반영할 수 있었습니다. 감사합니다!😁🙇

위에 의견주신 `when-thenReturn` 도 given 단계에 맞춰서 수정 반영했습니다.